### PR TITLE
Add config and updates for Llama 3.1

### PR DESCRIPTION
**Pull request by strickvl**
- [x] update config
- [x] update requirements
- [x] removed gradient checkpointing to make accelerate configs work again
- [x] update README to showcase the different configs we now have

### CODE REVIEW

File: llm-lora-finetuning/requirements.txt [hunk 1]
```diff
@@ -8,6 +8,9 @@ rouge_score
 nltk
 accelerate>=0.30.0
 urllib3<2
-zenml
+zenml>=0.62.0
 torch>=2.2.0
 sentencepiece
+pyOpenSSL==24.1.0
+protobuf
```
Comment (strickvl) [inline llm-lora-finetuning/requirements.txt:15]:
```suggestion
```

File: llm-lora-finetuning/requirements.txt [hunk 2]
```diff
@@ -8,6 +8,7 @@ rouge_score
 nltk
 accelerate>=0.30.0
 urllib3<2
-zenml
+zenml>=0.62.0
 torch>=2.2.0
 sentencepiece
+huggingface_hub
```
Comment (strickvl) [inline llm-lora-finetuning/requirements.txt:14]: This handles gated models

File: llm-lora-finetuning/steps/finetune.py [hunk 3]
```diff
@@ -121,7 +134,7 @@ def finetune(
             output_dir=output_dir,
             warmup_steps=warmup_steps,
             per_device_train_batch_size=per_device_train_batch_size,
-            gradient_checkpointing=True,
+            gradient_checkpointing=False,
```
Comment (strickvl) [inline llm-lora-finetuning/steps/finetune.py:137]: This allows accelerated configs to work again

**Review by safoinme — APPROVED** [review]
